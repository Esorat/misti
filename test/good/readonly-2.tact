fun test1(): Int {
    let a: Int = 20;
    let b: Int = 22;
    return b;
}
