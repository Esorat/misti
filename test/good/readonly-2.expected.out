test/good/readonly-2.tact:2:5:
  1 | fun test() {
> 2 |     let a: Int = 10; // warning: read-only variable
          ^
  3 |     repeat (a) { /* do nothing */ }
Read-only variable
Help: Consider creating a constant instead
See: https://nowarp.github.io/docs/misti/docs/detectors/ReadOnlyVariables
test/good/readonly-2.tact:3:13:
  2 |     let a: Int = 10; // warning: read-only variable
> 3 |     repeat (a) { /* do nothing */ }
                  ^
  4 |     repeat (a) { /* ... */ }
Unbounded Loop: The condition variable doesn't change within the loop
Help: Consider changing the variable within the loop to ensure it terminates
See: https://nowarp.github.io/docs/misti/docs/detectors/UnboundLoops
test/good/readonly-2.tact:4:13:
  3 |     repeat (a) { /* do nothing */ }
> 4 |     repeat (a) { /* ... */ }
                  ^
  5 | }
Unbounded Loop: The condition variable doesn't change within the loop
Help: Consider changing the variable within the loop to ensure it terminates
See: https://nowarp.github.io/docs/misti/docs/detectors/UnboundLoops